src/melody_extraction/util.py

```diff
@@ -0,0 +1,10 @@
+import numpy as np
+from numpy.typing import NDArray
+
+
+def mode(arr: NDArray[np.float16]) -> np.float64 | None:
```

Comment:
It could be nice for this to return a list of modes (in the event there is more than 1).